pennylane/grouping/pauli.py

```diff
@@ -0,0 +1,254 @@
+    """Generator function for the Pauli group.
+
+    This function is called by ``pauli_group`` in order to actually generate the
+    group elements. They are split so that the outer function can handle input
+    validation, while this generator is responsible for performing the actual
+    operations.
+
+    Args:
+        n_qubits (int): The number of qubits for which to create the group.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels
+            used in the Pauli word as keys, and unique integer labels as their values.
+            If no wire map is provided, wires will be labeled by integers between 0 and ``n_qubits``.
+
+    Returns:
+        .Operation: The next Pauli word in the group.
+    """
+
+    element_idx = 0
+
+    if not wire_map:
+        wire_map = {wire_idx: wire_idx for wire_idx in range(n_qubits)}
+
+    while element_idx < 4 ** n_qubits:
+        binary_string = format(element_idx, f"#0{2*n_qubits+2}b")[2:]
+        binary_vector = [float(b) for b in binary_string]
+        yield binary_to_pauli(binary_vector, wire_map=wire_map)
+        element_idx += 1
+
+
+def pauli_group(n_qubits, wire_map=None):
+    """Generate the :math:`n`-qubit Pauli group.
+
+    This function enables the construction of the :math:`n`-qubit Pauli group with no
+    storage involved.  The :math:`n`-qubit Pauli group has size :math:`4^n`,
+    thus it may not be desirable to construct it in full and store.
+
+    The order of iteration is based on the binary symplectic representation of
+    the Pauli group as :math:`2n`-bit strings. Ordering is done by converting
+    the integers :math:`0` to :math:`2^{2n}` to binary strings, and converting those
+    strings to Pauli operators using the ``binary_to_pauli`` method.
+
+    Args:
+        n_qubits (int): The number of qubits for which to create the group.
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels
+            used in the Pauli word as keys, and unique integer labels as their values.
+            If no wire map is provided, wires will be labeled by integers between 0 and ``n_qubits``.
+
+    Returns:
+        .Operation: The next Pauli word in the group.
+
+    **Example**
+
+    The ``pauli_group`` generator can be used to loop over the Pauli group as follows:
+
+    .. code-block:: python
+
+        from pennylane.pauli import pauli_group
+
+        n_qubits = 3
+
+        for p in pauli_group(n_qubits):
+            print(p)
+
+    The Pauli group in full can be obtained in full like so:
+
+    .. code-block:: python
+
+       full_pg = list(pauli_group(n_qubits))
+
+    The group can also be created using a custom wire map (if no map is
+    specified, a default map of label :math:`i` to wire ``i`` will be created).
+
+    .. code-block:: python
+
+        n_qubits = 3
+        wire_map = {'a' : 0, 'b' : 1, 'c' : 2}
+
+        for p in pauli_group(n_qubits, wire_map=wire_map):
+            print(p)
```

Comment:
If it makes sense, I recommend rewriting them using `pycon` syntax so that the output statements can be shown as well